packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -138,18 +128,25 @@ module.exports = ({ strapi }) => ({
   },
 
   findPage(opts, uid, populate) {
-    const params = { ...opts, populate: getBasePopulate(uid, populate) };
+    const params = { ...opts, populate: getDeepPopulate(uid, populate, { maxLevel: 1 }) };
 
     return strapi.entityService.findPage(uid, params);
   },
 
   findWithRelationCountsPage(opts, uid, populate) {
-    const counterPopulate = addCreatedByRolesPopulate(getCounterPopulate(uid, populate));
-    const params = { ...opts, populate: counterPopulate };
+    const counterPopulate = getDeepPopulate(uid, populate, { countMany: true, maxLevel: 1 });
+    const params = { ...opts, populate: addCreatedByRolesPopulate(counterPopulate) };
 
     return strapi.entityService.findWithRelationCountsPage(uid, params);
   },
 
+  findOneWithCreatorRolesAndCount(id, uid, populate) {
+    const counterPopulate = getDeepPopulate(uid, populate, { onlyMany: true, countMany: true });
```

Comment:
I'm not sure I understand why we don't populate nonMany relations ? Do they really need to be lazy loaded ? I don't see how this can really improve perf